cpp/src/lance/io/limit.h

```diff
@@ -0,0 +1,53 @@
+//  Copyright 2022 Lance Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#pragma once
+
+#include <arrow/array.h>
+#include <arrow/result.h>
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <tuple>
+
+namespace lance::io {
+
+/// Plan for Limit clause:
+///
+///   LIMIT value:int64 [OFFSET value:int64]
+///
+class Limit {
+ public:
+  Limit() = delete;
+
+  /// Construct a Limit Clause with limit, and optionally, offset.
+  explicit Limit(int64_t limit, int64_t offset = 0) noexcept;
+
+  /// Apply limit to the size of the length.
+  ///
+  /// \return a tuple of [offset, length]. Return [0, 0] to skip a chunk.
```

Comment:
The returned length is not the same as the input `length right? Can we make that clear?

Can you add what the return value means in the code comments? IIUC, if you have a Chunk c of length L, when you do Limit.Apply(L), the return value is saying that c[offset:offset+length] should be part of the Limit output right?

Reply:
Updated the comments.